src/Http.hs

```diff
@@ -17,13 +23,31 @@ import           Network.HTTP.Types
 import           Network.Wai.Handler.Warp
 import           Network.Socket
 
-start :: String -> IO (Bool, String) -> IO ()
-start socketName trigger = do
+socketName :: String
+socketName = ".autospec.sock"
+
+socketAddr :: SockAddr
+socketAddr = SockAddrUnix socketName
+
+newSocket :: IO Socket
+newSocket = socket AF_UNIX Stream 0
+
+withSocket :: (Socket -> IO a) -> IO a
+withSocket action = bracket newSocket sClose action
+
+withServer :: IO (Bool, String) -> IO a -> IO a
```

Comment:
Yes, I was about to hack this into `mockery`, but I stopped for two reasons:

 - We use it in production code here, `mockery` is meant as a test dependency only
 - It was not clear to me what the right abstraction would be

But in general I agree that we want functions in `mockery` that allow you to have a HTTP server mock with low overhead.